packages/ag-charts-community/src/chart/data/dataModel.ts

```diff
@@ -487,24 +469,25 @@ export class DataModel<
         return processedData as Grouped extends true ? GroupedData<D> : UngroupedData<D>;
     }
 
-    private valueGroupIdxLookup({ matchGroupIds, matchIds, matchScopes }: PropertySelectors) {
+    private valueGroupIdxLookup({ matchGroupIds, matchIds }: PropertySelectors) {
         return this.values
             .map((def, index) => ({ def, index }))
             .filter(({ def }) => {
                 if (matchGroupIds && (def.groupId == null || !matchGroupIds.includes(def.groupId))) {
                     return false;
                 }
-                if (matchIds && (def.id == null || !matchIds.includes(def.id))) {
-                    return false;
-                }
-                return !(matchScopes && (def.scopes == null || !matchScopes.some((s) => def.scopes?.includes(s))));
+                return !(
+                    matchIds &&
+                    (def.ids == null ||
+                        !matchIds.some(([matchScope, matchId]) =>
+                            def.ids?.some(([defScope, defId]) => defScope === matchScope && defId === matchId)
+                        ))
+                );
```

Comment:
I'd imagine this could be broken into a few simpler steps?

e.g.:
```
if (!matchIds) return true;
if (def.ids == null) return false;

return matchIds.some(([matchScope, matchId]) =>
  def.ids?.some(([defScope, defId]) => defScope === matchScope && defId === matchId)
));
```